### DISCUSSION TITLE
Expansion to Time Slot Options

### DISCUSSION THREAD

**Discussion by Muskehound** [chrisbenincasa/tunarr#956] [Ideas]
![Screenshot 2024-11-09 at 5 55 04 PM](https://github.com/user-attachments/assets/97c36fef-29c8-467e-a75b-251197ef0ffd)

With time slot scheduling, I feel there's a few more options that could be added in to enhance the experience which at the moment could only be done by crude attempts by the user.

Additional Dropdowns:

PreRoll
Using flex lists, the option to have a specific video (or a random selection from multiple) for a pre-show clip to be run. For example, adult swim have a wide array of stingers that would be perfect. While US broadcast TV doesn't use these so often, it was always very common in the UK to have the channel ident (which would last maybe 10 seconds) to play prior to any show. 

Order (Amendment)
In addition to the Next Episode/Shuffle options on the dropdown, the ability to start in a random spot in a shows lifespan and then move onto the next episode (before looping back around) would be neat so schedules don't start with every show airing it's pilot episode. Also shuffling the custom shows would be nice (Dizque has a few options for list ordering built into the Custom List screen itself, either works fine imo) 

Flex List
While the channel should have its default flex to rely on, this might not be appropriate for every show. For example, kids shows would have adverts for toys while shows airing at a later hour would contain more car commercials. Maybe using checkboxes would allow multiple flex lists if desired. It would also be neat to roll into informercials to pad the evening out so the next morning is nicely reset rather than using a time slot for this. 

(Not Pictured)

View/Refresh Show
I'm aware that automatic refreshing of shows is a complicated feature that's being worked on, in the meantime it would be nice to have an option next to the show to see what episodes have been brought in already (as it's difficult to keep track), with the option to bring in new additions. For custom lists, the ability to refresh the list to bring in new entries would be useful too. 

Schedule Ordering
In some circumstances, seeing the default order of items added to the schedule is perfect. Personally, most of the time I prefer to work with the schedule order itself as I'm often plugging in gaps that appear,  retiming etc, which is currently tricky to see as a whole. Also filtering to a specific day would be useful for weekly schedules which can get very long (336 entries if you schedule on a 30 minute basis)

**Comment by avalonMMXXII** [comment]
That is a very good idea, and very practical, I asked about this on reddit, but I am not sure if they are working on this yet. I know ETV already does this. Right now flex is a bit all over the place as you mentioned and the only way to really fix this is to make redirect stealth channels, but that gets very tedious (especially if you have multiple non-stealth channels)...there is still no real option to run a "pre-roll" or "end-roll" either...or a "time/day based flex video which could be used as an ident or promo of some kind"...so those are things that really need to be added to make flex videos more realistic. Especially on non on-demand channels.

**Comment by chrisbenincasa** [comment]
These are all great ideas. Sorry that I haven't had a chance to respond in full. I think some of these are doable now and some should probably wait until the next big upgrade which will involve library syncing (enabling us much more flexible show scheduling and updating, since we'll always have a full view of the library)

**Reply by chrisbenincasa** [reply]
Oh and by the way - the "order" suggestion is now available w/ the "Ordered Shuffle" option in both slot schedulers

**Reply by Muskehound** [reply]
Thanks Chris! I shuffle between my ongoing projects and was planning on rotating back to Tunarr as I could see in the discord there's been a ton of updates. Looks like a great time to jump back in!
